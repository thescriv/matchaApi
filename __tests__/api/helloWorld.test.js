const { startApi, stopApi } = require('../../src/api')
const { apiClient } = require('../apiClient')

let client

describe('HelloWorld API', () => {
  beforeAll(async () => {
    await startApi(3000)

    client = new apiClient(3000)
  })

  afterAll(async () => {
    await stopApi()
  })

  describe('GET /hello_world/classic_get', () => {
    test('do fetch getHelloWorld', async () => {
      const { body, status } = await client.getHelloWorld()

      expect({ body, status }).toMatchSnapshot()
    })
  })

  describe('POST /hello_world/classic_http', () => {
    test('do fetch postHelloWorld', async () => {
      const { body, status } = await client.postHelloWorld({
        name: 'Koa Setup'
      })

      expect({ body, status }).toMatchSnapshot()
    })
  })
})
