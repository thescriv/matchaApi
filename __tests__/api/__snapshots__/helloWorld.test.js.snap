// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld API GET /hello_world/classic_get do fetch getHelloWorld 1`] = `
Object {
  "body": Object {
    "message": "Hello World",
  },
  "status": 200,
}
`;

exports[`HelloWorld API POST /hello_world/classic_http do fetch postHelloWorld 1`] = `
Object {
  "body": Object {
    "message": "Hello World undefined",
  },
  "status": 200,
}
`;
